sdk/core/Azure.Core/src/Pipeline/ServicePointHelpers.cs

```diff
@@ -38,15 +38,23 @@ internal static class ServicePointHelpers
 
         public static void SetLimits(ServicePoint requestServicePoint)
         {
-            // Only change when the default runtime limit is used
-            if (requestServicePoint.ConnectionLimit == RuntimeDefaultConnectionLimit)
+            try
             {
-                requestServicePoint.ConnectionLimit = IncreasedConnectionLimit;
-            }
+                // Only change when the default runtime limit is used
```

Comment:
Same comment for the other if statements, or generalize to 
```suggestion
                // Only change these configuration values when the default value is used
```